components/browser/session/src/main/java/mozilla/components/browser/session/engine/EngineObserver.kt

```diff
@@ -64,6 +65,16 @@ internal class EngineObserver(
         }
     }
 
+    private fun isUrlSame(originalUrl: String, newUrl: String): Boolean {
+        val originalUri = Uri.parse(originalUrl)
+        val uri = Uri.parse(newUrl)
+
+        return uri.port == originalUri.port &&
+            uri.host == originalUri.host &&
+            uri.path?.trimStart('/') == originalUri.path?.trimStart('/') &&
+            uri.query == originalUri.query
+    }
```

Comment:
Eventually we could consider landing this as an extension method on `Uri` in `support-ktx`. But no need to do that now. :)